pystac_client/item_search.py

```diff
@@ -311,23 +313,34 @@ def get_parameters(self) -> Dict[str, Any]:
         if self.method == "POST":
             return self._parameters
         elif self.method == "GET":
-            params = deepcopy(self._parameters)
-            if "bbox" in params:
-                params["bbox"] = ",".join(map(str, params["bbox"]))
-            if "ids" in params:
-                params["ids"] = ",".join(params["ids"])
-            if "collections" in params:
-                params["collections"] = ",".join(params["collections"])
-            if "intersects" in params:
-                params["intersects"] = json.dumps(params["intersects"])
-            if "sortby" in params:
-                params["sortby"] = self._sortby_dict_to_str(params["sortby"])
-            if "fields" in params:
-                params["fields"] = self._fields_dict_to_str(params["fields"])
-            return params
+            return self._clean_params_for_get_request()
         else:
             raise Exception(f"Unsupported method {self.method}")
 
+    def _clean_params_for_get_request(self) -> Dict[str, Any]:
+        params = deepcopy(self._parameters)
+        if "bbox" in params:
+            params["bbox"] = ",".join(map(str, params["bbox"]))
+        if "ids" in params:
+            params["ids"] = ",".join(params["ids"])
+        if "collections" in params:
+            params["collections"] = ",".join(params["collections"])
+        if "intersects" in params:
+            params["intersects"] = json.dumps(params["intersects"])
+        if "sortby" in params:
+            params["sortby"] = self._sortby_dict_to_str(params["sortby"])
+        if "fields" in params:
+            params["fields"] = self._fields_dict_to_str(params["fields"])
+        return params
+
+    def url_with_parameters(self) -> str:
+        params = self._clean_params_for_get_request()
+        request = Request("GET", self.url, params=params)
+        url = request.prepare().url
+        if url is None:
+            raise ValueError("Could not construct a full url")
+        return urllib.parse.unquote(url)
```

Comment:
I unquote here to put `,` characters back into the URL, instead of `%2C`. It's an aesthetics thing that I don't feel too strongly about, but I do know folks like to see `,` in URLs instead of ASCII codes.